### PR TITLE
docs(button): add more comprehensive button examples

### PR DESCRIPTION
**Pull request by livvielin**
Screen Shot 2020-03-25 at 5 39 41 PM

### CODE REVIEW

File: src/components-examples/material/button/button-overview/button-overview-example.html [hunk 1]
```diff
@@ -1 +1,104 @@
...
+    <button mat-flat-button color="warn">Warn</button>
+    <button mat-flat-button disabled>Disabled</button>
+    <a mat-flat-button href="https://www.google.com/" target="_blank">Link</a>
+  </div>
+</section>
+<mat-divider></mat-divider>
+<section>
+  <div class="example-label">Icon</div>
+  <div class="example-button-row">
+    <button mat-icon-button aria-label="Example icon-button with a vertical three dot icon">
```
Comment (crisbeto) [inline src/components-examples/material/button/button-overview/button-overview-example.html]: nit: shouldn't this be just "icon button"?

File: src/dev-app/button/button-demo.html [hunk 2]
```diff
@@ -1,4 +1,9 @@
 <div class="demo-button">
+  <h4 class="demo-section-header">Overview</h4>
```
Comment (crisbeto) [inline src/dev-app/button/button-demo.html]: nit: can we move this under the existing code? Occasionally we add temporary code to these demos to test things out and it's easier when we don't have to scroll down after every refresh.

File: src/components-examples/material/button/button-overview/button-overview-example.html (hunk 1 above)
Comment (livvielin) [reply src/components-examples/material/button/button-overview/button-overview-example.html]: Fixed

File: src/dev-app/button/button-demo.html (hunk 2 above)
Comment (livvielin) [reply src/dev-app/button/button-demo.html]: Sounds good, moved to bottom

**Comment by jelbourn** [comment]
@livvielin what do you think of changing the icon and fab rows to evenly distribute whitespace between the buttons? I think that would look a tiny bit neater. It should be pretty simple to do with flexbox

**Comment by livvielin** [comment]
> @livvielin what do you think of changing the icon and fab rows to evenly distribute whitespace between the buttons? I think that would look a tiny bit neater. It should be pretty simple to do with flexbox

I think that seems better, but I did manually set width here and not sure that's the right approach. I think the icons look nice, but the FAB rows look kind of wonky just because of the size difference between the FAB and mini FAB (see updated screenshot). WDYT?

**Comment by livvielin** [comment]
> @livvielin what do you think of changing the icon and fab rows to evenly distribute whitespace between the buttons? I think that would look a tiny bit neater. It should be pretty simple to do with flexbox

I think that seems better, but I did manually set width here and not sure that's the right approach. I think the icons look nice, but the FAB rows look kind of wonky just because of the size difference between the FAB and mini FAB (see updated screenshot). WDYT?

**Comment by jelbourn** [comment]
I think it looks okay since it's still symmetrical; you could optionally wrap each button in a div that all have thew same width, and then center the button in that div, but I don't know if that's strictly necessary

**Comment by livvielin** [comment]
> I think it looks okay since it's still symmetrical; you could optionally wrap each button in a div that all have thew same width, and then center the button in that div, but I don't know if that's strictly necessary

Tried it out, and I think might look a bit neater that way

**Comment by jelbourn** [comment]
Awesome, thanks!

**Pull request merged**